### DISCUSSION TITLE
How to verifyTypedData off-chain?

### DISCUSSION THREAD

**Discussion by kostysh** [wevm/viem#580] [Question] [answered]
Is it possible to `verifyTypedData` (eip-712) off-chain without making an RPC call using `publicClient.verifyTypedData`?
I want to implement a simple testing script without the requirement to spin up a testing network.

It was easily possible with `ethers.js` where a local wallet account had both functions `signTypedData` and `verifyTypedData` as well.
On the opposite - using the `viem` it is possible to `sign` using local accounts but for verification, I need to create a public client that required a network setup.

**Comment by jxom** [comment, marked as answer]
Here you go! https://viem.sh/docs/utilities/verifyTypedData.html
